Automattic/woocommerce-payments
# Catch error when validating checkout fields

Fixes #6230 

#### Changes proposed in this Pull Request

When checking out with deferred intent UPE, credit card field validation occurs upon form submission. If an error occurs during validation, we don't exit from the `checkout()` function and `createStripePaymentMethod()` still runs and fails. This PR catches errors that occur during validation and exits the `checkout()` function, preventing `createStripePaymentMethod()` from running, ultimately preventing the cryptic error notice and message described in #6230.









#### Testing instructions

1. Ensure "Enable Split UPE checkout with deferred intent creation" is checked in your WCPay Dev Tools and ensure WooPay is deselected in your WCPay settings, so that the CC gateway uses the UPE instead of the legacy card element.
2. Add an item to cart and proceed to shortcode checkout.
3. With the credit card gateway, select "Use a new payment method", leave all the fields blank, and then hit "Place order".
4. Notice that the empty payment fields light up in red--which is what we want--but the page no longer scrolls up to reveal another cryptic message at the top of the page declaring, "paymentMethodObject.paymentMethod is undefined." or "Cannot read properties of undefined (reading 'id')".


-------------------


**Post merge**
